pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -343,6 +333,25 @@ if [[ -n $flake ]]; then
     fi
 fi
 
+# Re-execute nixos-rebuild from the Nixpkgs tree.
+if [[ -z $_NIXOS_REBUILD_REEXEC && -n $canRun && -z $fast ]]; then
+    if [[ -z $flake ]]; then
+        if p=$(nix-build --no-out-link --expr 'with import <nixpkgs/nixos> {}; config.system.build.nixos-rebuild' "${extraBuildFlags[@]}"); then
+            SHOULD_REEXEC=1
+        fi
+    else
+        if p=$(nix "${flakeFlags[@]}" build --no-link --json "$flake#$flakeAttr.config.system.build.nixos-rebuild" "${extraBuildFlags[@]}" "${lockFlags[@]}" | jq -r ".[0].outputs.out"); then
```

Comment:
`jq` might not be available, don't use it. I'd call nix with `--out-link "${tmpDir}/result"` and then read the path with `readlink -f "${tmpDir}/result"`, as is done later in the script in `nixFlakeBuild`.

Reply:
`jq` can be re-added to the derivation here: https://github.com/NixOS/nixpkgs/commit/c75bc3abc760a07eb7afc0be26eb1813a9867a84

Reply:
I'd prefer not adding unnecessary dependencies to a tool that might be used in various rescue situations.

Reply:
My point is that `nixos-rebuild` already dependent on `jq` on the past, so this is nothing new.

Reply:
BTW, I was the person that removed `jq` and the only reason I removed it was because the feature that was using it was removed because of an issue.

Reply:
Yes, I understood that. If others think that adding `jq` back and using it instead of `readlink` is the right thing to do, I'll accept it :-). In most situations, it will make no difference. In the end, `jq` is pretty lightweight (dependency-wise). My point was that if there are two possible solutions, one that needs a dependency and another that does not, I prefer the latter.

Reply:
Updated :+1: I've replaced the usage of `jq` with `readlink`